### ISSUE TITLE
[a11y]: ComposedModal - Passive Modal - screen reader focus in the dialog

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13889]
### Package

@carbon/react

### Package version

latest

### Automated testing tool and ruleset

QA testing

### Assistive technology

JAWS & VO

### Description

When the dialog is displayed, the screen reader focus is not set within the dialog. User has to bring the focus within the dialog by navigating using tab key. Along with that, when we press Insert + T to check the dialog label, jaws announces the browser title instead of announcing the dialog label and role.

### WCAG 2.1 Violation

2.4.3 Focus Order 

### Reproduction/example

https://ibm-studios.slack.com/archives/C054WK4ES5S/p1684934602201359?thread_ts=1684929656.513739&cid=C054WK4ES5S

### Steps to reproduce

User impact: When the focus is not set on a logical point in the page after a user action, it creates a great deal of confusion. The whole process of navigating back to the relevant element can be tedious. note this defect mainly impacts keyboard-only users and those with cognitive disabilities. 

Recommended fix: SR / Keyboard focus should be set within the dialog as soon as the dialog is displayed

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
